lib/AST/ASTContext.cpp

```diff
@@ -696,6 +696,7 @@ FuncDecl *ASTContext::getPlusFunctionOnString() const {
       if (auto type = dyn_cast<DECL_CLASS>(result)) { \
         auto params = type->getGenericParams(); \
         if (NUM_GENERIC_PARAMS == (params == nullptr ? 0 : params->size())) { \
+          getImpl().NAME##Decl = type; \
```

Comment:
!

Reply:
?

Reply:
This change implies that this has never been cached, ever, right? (Or at least since the last refactoring.

Reply:
It was being cached, but the control flow in the previous version made it a little unintuitive.

Reply:
Oh, whoops, I saw the partial diff from your previous version and thought it was a complete change. This makes sense.